apps/server/src/shared/controller/transformer/sanitize-html.transformer.ts

```diff
@@ -1,55 +1,196 @@
 import { NotImplementedException } from '@nestjs/common';
 import { Transform, TransformFnParams } from 'class-transformer';
-import sanitize from 'sanitize-html';
-
-type SanitizeDecoratorOptions = { keep: 'inline' | 'richtext' };
-
-// Note: tag names are not case-sensitive
-const INLINE_TAGS = ['b', 'i', 'em', 'strong', 'small', 's', 'u'];
-const HEADING_TAGS = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
-const LIST_TAGS = ['ul', 'li', 'ol', 'dl', 'dt', 'dd'];
-const PARAGRAPH_TAGS = ['p', 'pre', 'br', 'hr'];
-const TABLE_TAGS = ['table', 'tbody', 'td', 'tfoot', 'th', 'thead', 'tr', 'tr', 'td'];
-const OTHER_TAGS = ['a', 'img'];
-
-const getSanitizeHtmlOptions = (options?: SanitizeDecoratorOptions): sanitize.IOptions => {
-	let sanitizeHtmlOptions: sanitize.IOptions;
-
-	if (options?.keep === 'inline') {
-		sanitizeHtmlOptions = {
-			allowedTags: INLINE_TAGS,
-		};
-	} else if (options?.keep === 'richtext') {
-		sanitizeHtmlOptions = {
-			allowedTags: [...INLINE_TAGS, ...HEADING_TAGS, ...LIST_TAGS, ...PARAGRAPH_TAGS, ...TABLE_TAGS, ...OTHER_TAGS],
-			allowedAttributes: {
-				a: ['href', 'name', 'target'],
-				img: ['src', 'srcset', 'alt', 'title', 'width', 'height', 'loading'],
-			},
-		};
-	} else {
-		sanitizeHtmlOptions = { allowedTags: [], allowedAttributes: {} };
-	}
+import sanitize, { AllowedAttribute } from 'sanitize-html';
+import { InputFormat } from '@shared/domain';
+
+export type IInputFormatsConfig = {
+	allowedTags: string[]; // Note: tag names are not case-sensitive
+	allowedAttributes?: Record<string, AllowedAttribute[]>;
+};
+
+const inputFormatsSanitizeConfig: Record<string, IInputFormatsConfig> = {
+	PlainText: {
+		allowedTags: [],
+		allowedAttributes: {},
+	},
+
+	RichTextSimple: {
+		allowedTags: ['b', 'i', 'em', 'strong', 'small', 's', 'u'],
+		allowedAttributes: {},
+	},
+
+	RichText: {
+		allowedTags: [
+			// eslint-disable
+			'b',
+			'i',
+			'em',
+			'strong',
+			'small',
+			's',
+			'u',
+			'h1',
+			'h2',
+			'h3',
+			'h4',
+			'h5',
+			'h6',
+			'ul',
+			'li',
+			'ol',
+			'dl',
+			'dt',
+			'dd',
+			'p',
+			'pre',
+			'br',
+			'hr',
+			'table',
+			'tbody',
+			'td',
+			'tfoot',
+			'th',
+			'thead',
+			'tr',
+			'tr',
+			'td',
+			'a',
+			'img',
```

Comment:
what about the figure tag? 
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figure

Reply:
this is the old richtext that we had. Basically is the only used in the news until now. The list of tags for it, are the ones defined before, just in a big array instead of several.

For the other formats related to ck5 and ck4, we still have to update the list of tags.